docs/decisions/0017-generalized-containers.rst

```diff
@@ -0,0 +1,101 @@
+=======================================
+
+This section defines container members, their order, and relationships, covering flexible connections and support for draft and published states of their members.
+
+- The members of a container can be any type of publishable content. E.g., sections, subsections, units, components, and any other publishable thing. For more details on publishable content, see `PublishableEntity`_.
+- Members within a container are maintained in a specific order as an ordered list. E.g., components within a unit, or units within a subsection, are presented in a specific order.
+- Containers represent their content hierarchy through a structure, like Course > Section > Subsection > Unit > Component, which defines parent-child relationships at each level.
+- Containers support both pinned and unpinned references for its members, allowing members to be pinned to a specific version or set to reference its latest version. For instance, component V1 might be used in a unit instead of its latest version.
+- The latest state of a member can be referenced by setting its version to ``None``, which consists of the standard for a floating version.
+- A single member (publishable entity) can be shared by multiple containers, allowing for reuse of content across different containers. For instance, a component can be shared by multiple units.
+
+4. Container Version History
+============================
+
+This section defines the various lists of container's versions (author-defined, initial, and frozen) used  to track the history of changes made to a container, allowing to view past versions and changes over time.
+
+- Each container version holds different lists of members (author-defined, initial, and frozen) to support rollback operations and history tracking for the container.
+- The author-defined list is the list of members that the author has defined for the version of the container.
+- The author-defined list won't change for a specific container version even if its references get soft-deleted.
+- The initial list is a copy of the author-defined list that has all versions pinned as they were at the time the container version was created.
+- The initial list is immutable for a container version.
+- The frozen list refers to the list of members at the time when the next version of the container is created.
+- The author-defined list is used to show the content of a container version as the author specified it, the frozen list can be used for discard operations on a draft version and the initial-list is part of the history of evolution of the container.
+
+Let's say a course author creates a unit with three components, all using floating versions. Each component's latest version is V1. The author-defined list would include these three components, ordered as the author decided. The initial list would have the components pinned to V1, while the frozen list would be empty until we create the next version for the container.
```

Comment:
```suggestion
Let's say a course author creates a unit version with three components, all using floating versions. Each component's latest version is V1. The author-defined list would include these three components, ordered as the author decided. The initial list would have the components pinned to V1 since that's the components' versions for this moment in time, while the frozen list would be empty until we create the next version for the unit.
```